### PR TITLE
log TableName and RequestId with errors

### PR DESCRIPTION
**Pull request by danielhochman**
Example:

> `pynamodb.exceptions.PutError: Failed to put item: An error occurred (ConditionalCheckFailedException) on request (ee416189-1e4f-4911-aef3-1db0a835adf9) on table (MyTable) when calling the PutItem operation: The conditional request failed`

### CODE REVIEW

File: pynamodb/exceptions.py [hunk 1]
```diff
@@ -83,3 +85,19 @@ class TableDoesNotExist(PynamoDBException):
     def __init__(self, table_name):
         msg = "Table does not exist: `{0}`".format(table_name)
         super(TableDoesNotExist, self).__init__(msg)
+
+
+class VerboseClientError(botocore.exceptions.ClientError):
+    def __init__(self, error_response, operation_name, verbose_properties=None):
```
Comment (babldev) [inline pynamodb/exceptions.py]: `verbose_properties={}`?

File: pynamodb/exceptions.py [hunk 2]
```diff
@@ -83,3 +85,19 @@ class TableDoesNotExist(PynamoDBException):
...
+        """ Modify the message template to include the desired verbose properties """
+        if not verbose_properties:
+            verbose_properties = {}
+
+        self.MSG_TEMPLATE = (
+            'An error occurred ({{error_code}}) on request ({request_id}) '
```
Comment (babldev) [inline pynamodb/exceptions.py]: What are the double curly braces for?

Comment (danielhochman) [inline pynamodb/exceptions.py]: so they are passed onto the super's format(). calling format() on double curlies makes them single curlies

File: pynamodb/exceptions.py (hunk 1 above)
Comment (danielhochman) [inline pynamodb/exceptions.py]: dictionaries are mutable

**Comment by babldev** [comment]
👍

**Comment by danielhochman** [comment]
will cherry pick!

**Pull request closed without merging**